pkg/restore/import.go

```diff
@@ -196,32 +196,68 @@ func (importer *FileImporter) Import(file *backup.File, rewriteRules *RewriteRul
 					zap.Stringer("region", info.Region),
 					zap.Binary("startKey", startKey),
 					zap.Binary("endKey", endKey),
-					zap.Error(err1))
-				return err1
+					zap.Error(errDownload))
+				return errDownload
 			}
-			err1 = importer.ingestSST(downloadMeta, info)
-			// If error is `NotLeader`, update the region info and retry
-			for errors.Cause(err1) == errNotLeader {
-				log.Debug("ingest sst returns not leader error, retry it",
-					zap.Stringer("region", info.Region))
-				var newInfo *RegionInfo
-				newInfo, err1 = importer.metaClient.GetRegion(importer.ctx, info.Region.GetStartKey())
-				if err1 != nil {
-					break
+			ingestResp, errIngest := importer.ingestSST(downloadMeta, info)
+		ingestRetry:
+			for errIngest == nil {
+				errPb := ingestResp.GetError()
+				if errPb == nil {
+					// Ingest success
+					break ingestRetry
 				}
-				if !checkRegionEpoch(newInfo, info) {
-					err1 = errEpochNotMatch
-					break
+				var newInfo *RegionInfo
+				switch {
+				case errPb.NotLeader != nil:
+					// If error is `NotLeader`, update the region info and retry
+					if newLeader := errPb.GetNotLeader().GetLeader(); newLeader != nil {
```

Comment:
Move the `var newInfo *RegionInfo` inside the `case errPb.NotLeader != nil` please.